### PR TITLE
API provides possibility to create article

### PR DESCRIPTION
**Pull request by clalil**
# API provides possibility to create article
[PT-link](https://www.pivotaltracker.com/story/show/169265195)

## This PR contains
- Ability to create articles with images on the server side
- Request specs for happy and sad paths

## What we have learned
- How to use base64 in order to implement an attach image to articles functionality
- Further understanding of Rails and request specs

### CODE REVIEW

**Review by miyeaier — APPROVED** [review]

**Review by AlexSaxena — APPROVED** [review]

File: app/models/article.rb [hunk 1]
```diff
@@ -1,3 +1,6 @@
 class Article < ApplicationRecord
   validates_presence_of :title, :content
+  validates :title, length: { minimum: 3, maximum: 50 }
```
Comment (piavW) [inline app/models/article.rb]: Code repetition, either make it oneliner or remove the validates_presence_of and add presence validation to the separate attributes.

File: db/schema.rb [hunk 2]
```diff
@@ -10,18 +10,46 @@
...
+  end
+
   create_table "articles", force: :cascade do |t|
     t.string "title"
     t.text "content"
     t.datetime "created_at", precision: 6, null: false
     t.datetime "updated_at", precision: 6, null: false
   end
 
+  create_table "journalists", force: :cascade do |t|
```
Comment (piavW) [inline db/schema.rb]: why journalists?

**Review by piavW — CHANGES_REQUESTED** [review]
Please check comments

File: app/models/article.rb (hunk 1 above)
Comment (clalil) [reply app/models/article.rb]: Of course!

File: db/schema.rb (hunk 2 above)
Comment (clalil) [reply db/schema.rb]: Was left from before, we removed it.

**Review by piavW — APPROVED** [review]

**Pull request merged**